content/blog/2016/2016-06-29-from-freestyle-to-pipeline.adoc

```diff
@@ -0,0 +1,213 @@
+---
+layout: post
+title: "Migrating from chained Freestyle jobs to Pipelines"
+tags:
+- pipeline
+- infra
+author: rtyler
+---
+
+[NOTE]
+--
+This is a guest post by link:https;//github.com/rtyler[R. Tyler Croy], who is a
+long-time contributor to Jenkins and the primary contact for Jenkins project
+infrastructure. He is also a Jenkins Evangelist at
+link:http://cloudbees.com[CloudBees, Inc.]
+--
+
+For ages I have used the "Build After" feature in Jenkins to cobble together
+what one might refer to as a "pipeline" of sorts. The Jenkins project itself, a
+major consumer of Jenkins, has used these daisy-chained Freestyle jobs to drive
+a myriad of delivery pipelines in our infrastructure.
+
+One such "pipeline" helped drive the complex process of generating the pretty
+blue charts on
+link:http://stats.jenkins-ci.org/jenkins-stats/svg/svgs.html[stats.jenkins-ci.org].
+This statistics generation process primarily performs two major tasks, on rather
+large sets of data:
+
+. Generate aggregate monthly "census data."
+. Process the census data and create trend charts
+
+
+The chained jobs allowed us to resume the independent stages of the pipeline,
+and allowed us to run different stages on different hardware (different
+capabilities) as needed. Below is a diagram of what this looked like:
+
+image::/images/post-images/freestyle-to-pipeline-2016/freestyle-pipeline.png[role=center]
+
+
+The `infra_generate_monthly_json` would run periodically creating the
+aggregated census data, which would then be picked up by `infra_census_push`
+whose sole responsibility was to take census data and publish it to the
+necessary hosts inside the project's infrastructure.
+
+
+The second, semi-independent, "pipeline" would also run periodically. The
+`infra_statistics` job's responsibility was to use the census data, which had
```

Comment:
> "The infra_statistics job would then use..."

I shied away from this because that results in `would` being a bit repetitive for my tastes over the preceding few sentences